pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -695,6 +696,16 @@ func getServersConfig(cr *api.KieApp) ([]api.ServerTemplate, error) {
 			}
 
 			// Set replicas
+			// if the configuration use only one replica we increment
+			if strings.HasPrefix(string(cr.Spec.Environment), constants.RhdmPrefix) && len(cr.Spec.Objects.Servers) != 0 && cr.Spec.Objects.Servers[0].JbpmCluster && *serverReplicas == 1 {
```

Comment:
again, `len(cr.Spec.Objects.Servers)` is not required here are we're already in a loop for this object array. we know it's not zero at this point.

why are you using `cr.Spec.Objects.Servers[0].JbpmCluster` here instead of `serverSet.JbpmCluster` as we've already discussed and as you use below this?

Reply:
removed